README.md

```diff
@@ -91,12 +114,60 @@ jobs:
           name: GDExtension
           path: |
             ${{ github.workspace }}/bin/**
+```
 
+The above example is a lengthy one, so we will go through it action by action to see what is going on.
+
+In the `name: Checkout` step, we checkout the code.
+In the `name: 🔗 GDExtension Build` step, we are using the reusable action:
+```yml
+uses: godotengine/godot-cpp-template/.github/actions/build@main
+with:
+  platform: ${{ matrix.platform }}
+  arch: ${{ matrix.arch }}
+  float-precision: single
+  build-target-type: template_release
 ```
+with the parameters from the matrix.
+
+As a result of this step, the binaries will be built in the bin folder (As specified in the SConstruct file).
 
-## Usage - Signing
+Note: for macos, you will have to build the binary as a dylib in a my_lib.framework folder. The framework folder should also have a Resources folder with a file called Info.plist. Without this file, signing will fail.
 
-You will need:
+Note: for ios, the same should be as for macos, however the Info.plist file needs to be close to the dylib, instead of in a Resources folder (If this is not done, the build will fail to upload to appstore).
+
+So, in our case, the builds should be:
+
+```sh
+bin/libgdexample.macos.template_debug.framework/libEXTENSION-NAME.macos.template_release
+bin/libgdexample.ios.template_debug.framework/libEXTENSION-NAME.ios.template_release.arm64.dylib
+```
+
+Afterwards, you want to set in the `.gdextension` file the paths to the .framework folder, instead of the dylib (Note that for the dylib binary, the extension is not needed, you could have a file without any extension and it would still work).
```

Comment:
```suggestion
Afterwards, you want to set in the `.gdextension` file the paths to the `.framework` folder, instead of the `.dylib` file (Note that for the `.dylib` binary, the extension is not needed, you could have a file without any extension and it would still work).
```